src-extras/Database/LSMTree/Extras/Generators.hs

```diff
@@ -196,8 +219,8 @@ shrinkWriteBuffer ::
   => (k -> [k])
   -> (v -> [v])
   -> (blob -> [blob])
-  -> WriteBuffer k v blob
-  -> [WriteBuffer k v blob]
+  -> WriteBuffer
+  -> [WriteBuffer]
```

Comment:
We could also monomorphise `gen/shrinkWriteBuffer`, so it would take a `SerialisedKey -> [SerialisedKey]` argument etc., remove `fromKOps` and `toKOps` and then in the instances do `shrinkWriteBuffer arbitrary _ _` and `shrinkWriteBuffer (fmap serialiseKey . shrink @k . deserialiseKey) _ _` respectively.

Does the same thing in the end, but fits the untyped/typed split and might be cleaner. Do think it would be worth it?

Reply:
I've instead changed `genWriteBuffer`/`shrinkWriteBuffer` to `genTypedWriteBuffer`/`shrinkTypedWriteBuffer`. `Arbitrary WriteBuffer` is now a special case of `Arbitrary (TypedWriteBuffer k v blob)`. Basically I've flipped the relationship around